src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -429,33 +434,42 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
   /**
    * Converts feedback question to the question edit form model.
    */
-  private getQuestionEditFormModel(feedbackQuestion: FeedbackQuestion): QuestionEditFormModel {
-    return {
-      feedbackQuestionId: feedbackQuestion.feedbackQuestionId,
+  private getQuestionEditFormModel(feedbackQuestion: FeedbackQuestion): Promise<QuestionEditFormModel> {
```

Comment:
I am still not sure why you had to use a Promise here?  For our frontend code, we normally have been using callbacks (e.g. subscribe) to get the result of doing a async operation. 

Are you using this to check the response count after a question has been edited? Where is the warning message you showed on the home screen coming from?

Reply:
Maybe you want to use Observable instead.

However I think a better way to do this is to separate the call to get `isQuestionHasResponses` into another function. `getQuestionEditFormModel` will remain sync.

Reply:
The warning message is shown in the page where an instructor attempts to edit a feedback session's questions. If any question has responses, it is displayed for that specific question.